common/lib/xmodule/xmodule/js/src/video/03_video_player.js

```diff
@@ -806,8 +798,9 @@ function (HTML5Video, Resizer) {
 
     function updatePlayTime(time, skip_seek) {
         var videoPlayer = this.videoPlayer,
-            duration = this.videoPlayer.duration(),
+            duration = (this.config.endTime !== null) ? this.config.endTime : this.videoPlayer.duration(),
```

Comment:
Here's my take on the convention as it is right now:

`null` is the default value assigned for an erroneous endTime. It specifies that the endTime error is invalid (aka erroneous), rather than it's a missing field. The backend enforces that this will not be a "missing value", and imo making it "undefined" would change the semantics somewhat. The "undefined" check inside `00_sjson.js` is for arity; the checks in other files are for "erroneous" values of endTime.

Not sure if that makes sense / is overly complicated.